E1_coldflow/brain_utils.h

```diff
@@ -151,31 +182,49 @@ valveInfo valves[numValves] = {
   {"Launch Rocket", 28, &(Solenoids::LAUNCH), &(Solenoids::endBurn), &(Solenoids::getAllStates)},
 };
 
-void sensorReadFunc(int id) {
-  switch (id) {
-    case 0:
-      //Thermocouple::setSensor(0);
-      Ducers::readTemperatureData(farrbconvert.sensorReadings);
-      farrbconvert.sensorReadings[1] = tempController::controlTemp(farrbconvert.sensorReadings[0]);
-      farrbconvert.sensorReadings[2] = -1;
-      break;
-    case 1:
-      Ducers::readAllPressures(farrbconvert.sensorReadings);
-      break;
-    case 2:
-      batteryMonitor::readAllBatteryStats(farrbconvert.sensorReadings);
-      break;
-    case 4:
-      Thermocouple::Cryo::readCryoTemps(farrbconvert.sensorReadings);
-      //farrbconvert.sensorReadings[1]=0;
-      farrbconvert.sensorReadings[2]=0;
-      farrbconvert.sensorReadings[3]=0;
-      farrbconvert.sensorReadings[4]=-1;
-      break;
-    default:
-      Serial.println("some other sensor");
-      break;
+
+
+//void sensorReadFunc(int id) {
+//  switch (id) {
+//    case 0:
+//      readTemp();
+//      break;
+//    case 1:
+//      Ducers::readAllPressures(farrbconvert.sensorReadings);
+//      break;
+//    case 2:
+//      batteryMonitor::readAllBatteryStats(farrbconvert.sensorReadings);
+//      break;
+//    case 4:
+//      Thermocouple::Cryo::readCryoTemps(farrbconvert.sensorReadings);
+//      break;
+//    default:
+//      Serial.println("some other sensor");
+//      break;
+//  }
+//}
+
+void readTemp() {
+  Ducers::readTemperatureData(farrbconvert.sensorReadings);
+  farrbconvert.sensorReadings[1] = tempController::controlTemp(farrbconvert.sensorReadings[0]);
+  farrbconvert.sensorReadings[2] = -1;
+}
+
+/*
+ * Calls the corresponding method for this valve with the appropriate
+ * action in solenoids.h
+ */
+void take_action(valveInfo *valve, sensorInfo *sensor, int action) {
+  if (action == 1) {
+    valve->openValve();
+  } else if (action == 0) {
+    valve->closeValve();
+  } else if (action == 2) {
+    //call the reading function
+    sensor->readData(farrbconvert.sensorReadings);
```

Comment:
So the reason we had the switch case was that this never worked in the past. This needs to be tested more in an isolated way w/ minimal sensors.
And this looks good for requesting specific data, but I'm not sure I see where you handle the regular data. There will still be certain data that we want to just stream w/out receiving a command for.